_posts/2022-06-30-My first-take-on-KMM-Kotlin-Multiplatform-Mobile.md

```diff
@@ -0,0 +1,113 @@
+---
+layout: post
+title: My first take on KMM - Kotlin Multiplatform Mobile
+author: michal.guspiel, denzilferreira
+date: 2022-06-30 16:00:00 +0200
+excerpt: As for a first project and practice assignment in Solita I've had explored new technology - KMM and created Dev Notary, a simple application that lets users create, modify and share notes between other users.
+
+tags:
+  - KMM
+  - Android
+  - Kotlin
+  - Mobile
+  - Multiplatform
+  - iOS
+  - Swift
+  - Dev Academy
+---
+
+# Preface
+
+In order to finish the Dev Academy, I've had to develop an application as my practice assignment. I've been provided a mentor - Denzil Ferreira, whom I was guided by throughout the process. The purpose of practice assignment is to provide to a freshly recruited junior developers a safe environment, in which they can work on a small feature of an application or simple application of their own. Moreover, thanks to mentor guidance development gets done in a very proffessional fashion and is a great way to kickoff start for a new software designer.
+
+On the initial project kick-off meeting, my mentor and I have decided that to mix things up a little I should develop the app in a technology that I am not super familiar with. After going over several different possibilities, we settled with Kotlin Multiplatform Mobile (KMM). Some of the reasons behind this decision were: 1) my familiarity with Native Android Development and 2) KMM is an exciting and emerging technology worth exploring for a streamlined mobile development team.
+
+# Kotlin Multiplatform Mobile
+
+[Kotlin Multiplatform Mobile](https://kotlinlang.org/lp/mobile/) (KMM) is a tool developed by JetBrains to simplify iOS and Android app development. Kotlin Multiplatform empowers cross-platform (e.g., mobile, desktop, web) and yet also supports native- development on such platforms. This article focuses on the Mobile side of things.
+
+## Slicing apps into layers of concern
+
+The way to create an applications with KMM is to develop one shared codebase for business logic and then separate codebases for each platform specific features. Separate native codebases are usually for UI, and occasional depending on a case for platform related functionality. This approach doesn’t reduce native performance as Kotlin is compiled to native execution code. KMM saves teams' time and effort as you do not have to write redundant business logic in both apps, nor coordinate between teams for feature parity.
```

Comment:
The way to create -> To create

"Separate native codebases are usually for UI, and occasional" -> seems that there is something missing here. perhaps "...occasionally more, depending..." 

"nor coordinate" -> nor having to coordinate